fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordKeyCursorTest.java

```diff
@@ -0,0 +1,270 @@
+    public void testIterateRecordsMissingSplit(int splitNumber, UseContinuations useContinuations, int formatVersion) throws Exception {
+        boolean splitLongRecords = true;
+
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> savedRecords = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete a split
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // If operating on the short record, #0 is the only split
+            // If operating on the long record, splits can be 1,2,3
+            // Use splitNumber to decide which record to operate on.
+            // Record #1 in the saved records is a short record, #16 is a long (split) record
+            int recordNumber = (splitNumber == 0) ? 1 : 16;
+            final Tuple pkAndSplit = savedRecords.get(recordNumber).getPrimaryKey().add(splitNumber);
+            byte[] split = store.recordsSubspace().pack(pkAndSplit);
+            store.ensureContextActive().clear(split);
+            commit(context);
+        }
+
+        // Scan records
+        ScanProperties scanProperties = getScanProperties(useContinuations);
+        final List<Tuple> actualKeys;
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            actualKeys = scanKeys(store, scanProperties, useContinuations != UseContinuations.NONE);
+            commit(context);
+        }
+        List<Tuple> expectedKeys;
+        // When format version is 3 and the record is a short record, deleting the only split will make the record disappear
+        if ((splitNumber == 0) && (formatVersion == 3)) {
+            expectedKeys = IntStream.range(1, 101).filter(i -> i != 2).boxed().map(Tuple::from).collect(Collectors.toList());
+        } else {
+            expectedKeys = IntStream.range(1, 101).boxed().map(Tuple::from).collect(Collectors.toList());
+        }
+        assertEquals(expectedKeys, actualKeys);
+    }
+
+    public static Stream<Arguments> splitAndContinuation() {
+        return Stream.of(true, false)
+                .flatMap(
+                        split -> Arrays.stream(UseContinuations.values())
+                                .map(useContinuations -> Arguments.of(split, useContinuations)));
+    }
+
+    @ParameterizedTest(name = "testIterateRecordsMissingVersion [splitLongRecords = {0}, useContinuations = {1}]")
+    @MethodSource("splitAndContinuation")
+    void testIterateRecordsMissingVersion(boolean splitLongRecords, UseContinuations useContinuations) throws Exception {
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> savedRecords = saveRecords(splitLongRecords, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION, hook);
```

Comment:
You should probably also test this with `SAVE_UNSPLIT_WITH_SUFFIX_FORMAT_VERSION` and `RECORD_COUNT_KEY_ADDED_FORMAT_VERSION`

Reply:
For version 3 and 5 we don't have a version split, so would it make sense to remove it?